app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,34 @@
+class GroupsController < ApplicationController
+  before_action :authenticate_user!, except: [:index]
+
+  def index
+    return unless current_user
+
+    @user = current_user
+    @groups = @user.groups.includes(:entities)
+  end
+
+  def show
+    @group = Group.find(params[:id])
+    @entities = @group.entities.order('created_at DESC')
+  end
+
+  def new
+    @group = Group.new
+  end
+
+  def create
+    @group = Group.new(group_params)
+    @group.user_id = current_user.id
+
+    render :new unless @group.save
+
+    redirect_to groups_path
```

Comment:
- [ ] If I try to create a category without adding any link the app crashes, you can either validate that the user must enter a link or simply make it possible to continue without adding it. Because  it is not very user-friendly when the app crashes